src/GraphQLParser/AST/GraphQLIntValue.cs

```diff
@@ -8,13 +11,72 @@ namespace GraphQLParser.AST;
 [DebuggerDisplay("GraphQLIntValue: {Value}")]
 public class GraphQLIntValue : GraphQLValue
 {
+    private ROM _value;
+    private object? _number;
+
     /// <inheritdoc/>
     public override ASTNodeKind Kind => ASTNodeKind.IntValue;
 
     /// <summary>
     /// Integer value represented as <see cref="ROM"/>.
     /// </summary>
-    public ROM Value { get; set; }
+    public ROM Value
+    {
+        get => _value;
+        set
+        {
+            _value = value;
+            _number = null;
+        }
+    }
+
+    /// <summary>
+    /// Integer value represented as <see cref="int"/>, <see cref="long"/>, <see cref="decimal"/> or <see cref="BigInteger"/>.
+    /// <br/>
+    /// This property allocates the string on the heap on first access
+    /// and then caches it as long as <see cref="Value"/> does not change.
+    /// </summary>
+    private /*public*/ object TypedValue //TODO: ??? no typed value :(
```

Comment:
@Shane32 Please pay attention to `IntValue` and `FloatValue`. I did so, because the other options were even worse.

Reply:
`Parse` methods were ported from the main repo.

Reply:
Reverted